src/main/java/ch/kalunight/zoe/service/infochannel/TreatPlayerWorker.java

```diff
@@ -0,0 +1,423 @@
+      updateRankChannelMessage(playerToUpdate, leagueAccount, currentGameDb, lastRank);
+    }
+  }
+
+  private void updateRankChannelMessage(DTO.Player player, DTO.LeagueAccount leagueAccount, DTO.CurrentGameInfo currentGameDb,
+      LastRank lastRank) {
+    CurrentGameInfo gameOfTheChange = currentGameDb.currentgame_currentgame;
+
+    if(gameOfTheChange.getGameQueueConfigId() == GameQueueConfigId.SOLOQ.getId()) {
+
+      RankedChannelLoLRefresher rankedRefresher = 
+          new RankedChannelLoLRefresher(rankChannel, lastRank.lastRank_soloqSecond, lastRank.lastRank_soloq,
+              gameOfTheChange, player, leagueAccount, server);
+      ServerData.getRankedMessageGenerator().execute(rankedRefresher);
+
+    }else if(gameOfTheChange.getGameQueueConfigId() == GameQueueConfigId.FLEX.getId()) {
+
+      RankedChannelLoLRefresher rankedRefresher = 
+          new RankedChannelLoLRefresher(rankChannel, lastRank.lastRank_flexSecond, lastRank.lastRank_flex,
+              gameOfTheChange, player, leagueAccount, server);
+      ServerData.getRankedMessageGenerator().execute(rankedRefresher);
+    }
+
+  }
+
+
+  private void generateText(Map<DTO.LeagueAccount, CurrentGameInfo> accountsWithGame, List<DTO.LeagueAccount> accountNotInGame) throws SQLException {
+    
+    if(accountsWithGame.isEmpty()) {
+
+      if(serverConfig.getInfopanelRankedOption().isOptionActivated()) {
+
+        if(accountNotInGame.size() == 1) {
+
+          LeagueAccount leagueAccount = accountNotInGame.get(0);
+
+          getTextInformationPanelRankOption(infochannelMessage, player, leagueAccount, false);
+        }else if (accountNotInGame.size() > 1){
+
+          infochannelMessage.append(String.format(LanguageManager.getText(server.serv_language, "infoPanelRankedTitleMultipleAccount"), player.getUser().getAsMention()) + "\n");
+
+          for(DTO.LeagueAccount leagueAccount : accountNotInGame) {
+
+            getTextInformationPanelRankOption(infochannelMessage, player, leagueAccount, true);
+          }
+        }
+
+      } else {
+        notInGameWithoutRankInfo(infochannelMessage, player);
+      }
+    }else if (accountsWithGame.size() == 1) {
+      Entry<DTO.LeagueAccount, CurrentGameInfo> entry = accountsWithGame.entrySet().iterator().next();
+      infochannelMessage.append(player.getUser().getAsMention() + " : " 
+          + InfoPanelRefresherUtil.getCurrentGameInfoStringForOneAccount(entry.getKey(), entry.getValue(), server.serv_language) + "\n");
+    }else {
+      infochannelMessage.append(player.getUser().getAsMention() + " : " 
+          + LanguageManager.getText(server.serv_language, "informationPanelMultipleAccountInGame") + "\n"
+          + InfoPanelRefresherUtil.getCurrentGameInfoStringForMultipleAccounts(accountsWithGame, server.serv_language));
+    }
+  }
+
+  private void getTextInformationPanelRankOption(final StringBuilder stringMessage, DTO.Player player,

```

Comment:
Codacy found an issue: [The method 'getTextInformationPanelRankOption(StringBuilder, DTO.Player, DTO.LeagueAccount, boolean)' has an NPath complexity of 960, current threshold is 200](https://app.codacy.com/manual/KaluNight/Zoe-Discord-Bot/pullRequest?prid=6283923)